src/debtManagers/VariableDebtManager.sol

```diff
@@ -0,0 +1,132 @@
+//SPDX-License-Identifier: Unlicensed
+pragma solidity ^0.8.20;
+
+interface IDebtManager {
+    function increaseDebt(address user, uint amount) external;
+    function decreaseDebt(address user, uint amount) external returns(uint);
+    function debt(address market, address user) external view returns (uint);
+    function marketDebt(address market) external view returns(uint);
+}
+
+interface IDbrAMM {
+    function burnDbr(uint exactDolaIn, uint exactDbrBurn) external returns (uint dolaIn);
+}
+
+interface IHelper {
+    function dolaNeededForDbr(uint dbrOut) external view returns (uint dolaIn);
+}
+
+interface IDBR {
+    function markets(address) external view returns (bool);
+}
+
+interface IDOLA {
+    function approve(address, uint) external returns (bool);
+    function mint(address, uint) external;
+}
+
+
+contract VariableDebtManager is IDebtManager {
+
+    IDbrAMM public immutable amm; //TODO: Doesn't necessarily have to be immutable. Can consider making mutable.
+    IDBR public immutable dbr;
+    IDOLA public immutable dola;
+    IHelper public helper; //Should this functionality be a part of the AMM?
```

Comment:
If we dump the invariant model in favor of a simple price function then yes a helper wouldn't be needed